src/main/java/io/cdap/plugin/http/source/common/BaseHttpSourceConfig.java

```diff
@@ -774,7 +774,9 @@ public void validate() {
       String reasonFormat = String.format("page format is '%s'", getFormat());
 
       if (getFormat().equals(PageFormat.JSON) || getFormat().equals(PageFormat.XML)) {
-        assertIsSet(getResultPath(), PROPERTY_RESULT_PATH, reasonFormat);
+        if (!getFormat().equals(PageFormat.JSON)) {
+          assertIsSet(getResultPath(), PROPERTY_RESULT_PATH, reasonFormat);
```

Comment:
In what cases can resultPath be unset?

Reply:
resultPath can be unset in the cases where the top level is just a JSONArray without a path. 



> Can you please list the test cases you ran to validate these changes?
> 
> From the JIRA description:
> 
> ```
>  if it returns an array of objects, the source should return multiple records, one for each object in the array. 
> Schema should be set based on this logic, and pagination should also work similarly.
> ```
> 
> Is this handled in this PR? If yes, can you please point me to the part of PR that handles splitting into multiple records/pagination? Thanks.


The pagination will work just fine when the top level will be JSONObject with path included. 

Here you have the part of the PR that handles the pagination: 
https://github.com/data-integrations/http/blob/67b752d46aa01f0171a1e90f5ff30a37738897a9/src/main/java/io/cdap/plugin/http/source/common/pagination/page/JsonPage.java#L196-L205

In the cases where the top level is _JSONArray_, the pagination for types such as **Link in response body** and **Token in response body** makes no sense for me. 

Please correct me if my assumption is wrong.

Reply:
I see, thanks